src/com/owncloud/android/ui/activity/FileDisplayActivity.java

```diff
@@ -497,6 +509,12 @@ public boolean onPrepareOptionsMenu(Menu menu) {
     public boolean onCreateOptionsMenu(Menu menu) {
         MenuInflater inflater = getMenuInflater();
         inflater.inflate(R.menu.main_menu, menu);
+        mOptionsMenu = menu;
+
+        MenuItem menuItem = mOptionsMenu.findItem(R.id.action_switch_view);
```

Comment:
Done for the menuItem.
As I recall correctly it is important to have the mOptionMenu updated.